packages/react-mutation-mapper/src/util/CivicUtils.ts

```diff
@@ -0,0 +1,224 @@
+import * as _ from 'lodash';
+import CivicDataFetcher from '../store/CivicDataFetcher';
+import {
+    ICivicEntry,
+    ICivicGene,
+    ICivicGeneData,
+    ICivicVariant,
+    ICivicVariantData,
+} from '../model/Civic';
+import { Mutation } from '../model/Mutation';
+
+export type MutationSpec = {
+    gene: { hugoGeneSymbol: string };
+    proteinChange: string;
+};
+
+const civicClient = new CivicDataFetcher();
+
+enum AlterationType {
+    AMPLIFICATION = 'AMPLIFICATION',
+    DELETION = 'DELETION',
+}
+
+/**
+ * Asynchronously adds the given variant from a gene to the variant map specified.
+ */
+function addCivicVariant(
```

Comment:
I kind of hate this function since it hides the "side effects".

Reply:
i see that you are using this multiple times.  maybe change the name of this function to lookupCivicVariantAndAdd ToMap